catalog/async/catalog-async-data/src/main/java/org/codice/ddf/catalog/async/data/impl/LazyProcessResourceImpl.java

```diff
@@ -94,16 +94,16 @@ public LazyProcessResourceImpl(String metacardId, Supplier<Resource> resourceSup
   @Override
   public synchronized InputStream getInputStream() throws IOException {
     loadResource();
-    if (inputStreamDataCache == null) {
+    if (resourceDataCache == null) {
       if (inputStream == null) {
         throw new IOException(String.format("Tried to get input stream for %s but was null", name));
       }
-      inputStreamDataCache =
-          new TemporaryFileBackedOutputStream(FILE_BACKED_OUTPUT_STREAM_THRESHOLD);
-      IOUtils.copyLarge(inputStream, inputStreamDataCache);
-      streamCloser.register(inputStreamDataCache);
+      resourceDataCache = new TemporaryFileBackedOutputStream(FILE_BACKED_OUTPUT_STREAM_THRESHOLD);
+      IOUtils.copyLarge(inputStream, resourceDataCache);
```

Comment:
❗️ If the copy throws an IOException, the stream is leaked (see comment on `close()` below which could take care of that also.